src/main/java/io/moderne/organizations/OrganizationStructureService.java

```diff
@@ -38,69 +41,52 @@ public Map<String, OrganizationRepositories> readOrganizationStructure() {
 
-        final Map<String, String> idToNameMapping = readIdToNameMapping();
+        CsvMapper csvMapper = new CsvMapper();
+        CsvSchema bootstrapSchema = CsvSchema.emptySchema()
+                .withComments()
+                .withHeader();
 
-        InputStream inputStream;
-        try {
-            if (reposCsvPath == null) {
-                inputStream = new ClassPathResource(DEFAULT_REPOS_CSV).getInputStream();
-            } else {
-                inputStream = new FileInputStream(reposCsvPath.toFile());
-            }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        File inputFile;
+        if (reposCsvPath == null) {
+            inputFile = new File(DEFAULT_REPOS_CSV);
+        } else {
+            inputFile = reposCsvPath.toFile();
         }
 
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
-            reader.readLine(); // skip header
-            reader.lines()
-                    .forEach(line -> {
-                        String[] fields = line.split(",", Integer.MAX_VALUE);
-                        if (fields.length < 3) {
-                            throw new IllegalStateException("repos.csv lines should have at least 1 organization");
+        try (MappingIterator<ReposCsvRow> iterator = csvMapper.readerFor(ReposCsvRow.class)
+                .with(bootstrapSchema)
+                .readValues(inputFile)) {
+            while (iterator.hasNextValue()) {
+                ReposCsvRow row = iterator.nextValue();
+                RepositoryInput repository = determineRepository(row.getCloneUrl(), row.getBranch());
+
+                Queue<String> orgNames = new LinkedList<>(row.getOrganizations());
+                if (orgNames.isEmpty()) {
+                    throw new IllegalStateException("repos.csv lines should have at least 1 organization");
+                } else if (orgNames.contains("")) {
+                    throw new IllegalStateException("Invalid organization \"\" for %s".formatted(row.getCloneUrl()));
+                }
+
+                boolean first = true;
+                do {
+                    String organization = orgNames.poll();
+                    String parent = orgNames.peek();
+                    OrganizationRepositories organizationRepositories = organizations.compute(organization, (k, v) -> {
+                        if (v == null) {
+                            v = new OrganizationRepositories(organization, organization, new LinkedHashSet<>(), List.of(CommitOption.values()), parent);
                         }
-                        String cloneUrl = fields[0].trim();
-                        String branch = fields[1].trim();
-                        RepositoryInput repository = determineRepository(cloneUrl, branch);
-                        if (repository == null) {
-                            if (scmConfiguration.isAllowMissingScmOrigins()) {
```

Comment:
I was uncertain if we wanted to conserve this logic or not, but we this can be added back in

Reply:
I'll put this back in, for now and can remove if required

Reply:
yeah, we want to keep this logic.